web/reNgine/tasks.py

```diff
@@ -1351,11 +1351,22 @@ def fetch_endpoints(
 
         elif tool == 'hakrawler':
             logger.info('Running hakrawler')
-            hakrawler_command = '{} | hakrawler | grep -Eo {} > {}/urls_hakrawler.txt'.format(
-                input_target,
-                valid_url_of_domain_regex,
-                results_dir
-            )
+            if subdomain:
```

Comment:
This is not necessary as from Line 1344 to 1350 it is already being done

```
if tool == 'gauplus' or tool == 'hakrawler' or tool == 'waybackurls':
            if subdomain:
                subdomain_url = subdomain.http_url if subdomain.http_url else 'https://' + subdomain.name
                input_target = 'echo {}'.format(subdomain_url)
            elif scan_type == 'deep' and domain:
                input_target = 'cat {}'.format(sorted_subdomains_path)
            else:
                input_target = 'echo {}'.format(domain_name)
        if tool == 'gauplus':
            logger.info('Running Gauplus')
            gauplus_command = '{} | gauplus --random-agent | grep -Eo {} > {}/urls_gau.txt'.format(
                input_target,
                valid_url_of_domain_regex,
                results_dir
            )
            logger.info(gauplus_command)
            os.system(gauplus_command)
```

Reply:
hakrawler  in normal mode not working we need  " https or http"